docs/src/man/categorical.md

```diff
@@ -91,57 +91,90 @@ julia> cv = compress(cv)
 
 ```
 
-Often, you will have factors encoded inside a DataFrame with `Array` columns instead of `CategoricalArray` columns. You can do conversion of a single column using the `categorical` function:
-
-```jldoctest categorical
-julia> cv = categorical(v)
-6-element CategoricalArrays.CategoricalArray{String,1,UInt32}:
- "Group A"
- "Group A"
- "Group A"
- "Group B"
- "Group B"
- "Group B"
-
-```
-
-Or you can edit the columns of a `DataFrame` in-place using the `categorical!` function:
+Often, you will have factors encoded inside a DataFrame with `Array` columns instead of
+`CategoricalArray` columns. You can convert one or more columns of the DataFrame using the
+`categorical!` function, which modifies the input DataFrame in-place.
 
 ```jldoctest categorical
 julia> using DataFrames
 
-julia> df = DataFrame(A = [1, 1, 1, 2, 2, 2],
-                      B = ["X", "X", "X", "Y", "Y", "Y"])
+julia> df = DataFrame(A = ["A", "B", "C", "D", "D", "A"],
+                             B = ["X", "X", "X", "Y", "Y", "Y"])
 6×2 DataFrames.DataFrame
 │ Row │ A │ B │
 ├─────┼───┼───┤
-│ 1   │ 1 │ X │
-│ 2   │ 1 │ X │
-│ 3   │ 1 │ X │
-│ 4   │ 2 │ Y │
-│ 5   │ 2 │ Y │
-│ 6   │ 2 │ Y │
+│ 1   │ A │ X │
+│ 2   │ B │ X │
+│ 3   │ C │ X │
+│ 4   │ D │ Y │
+│ 5   │ D │ Y │
+│ 6   │ A │ Y │
+
+julia> allcols = deepcopy(df); bothcols = deepcopy(df); onecol = deepcopy(df)
```

Comment:
It's confusing to do all that setup. Maybe just show first that you can specify a single column, and then call the function without an argument and show that the other column was also converted? Then people can read the docstring to find out more.